tests/format/test_numbers.py

```diff
@@ -0,0 +1,220 @@
+from owid.datautils.format.numbers import (
+    Number,
+    NumberWithSeparators,
+    NumberWithWords,
+    num_to_str,
+    remove_multiple_whitespaces,
+    format_number,
+)
+from pytest import raises
+
+
+class TestNumberWithSeparators:
+    separators = [",", ".", " "]
+    numbers_valid = ["1,000,000", "1,234", "123,032", "21,000,100,231"]
+    numbers_valid_corrected = [1000000, 1234, 123032, 21000100231]
+    numbers_wrong = ["1,", "1,0", "2,01", "1234,000"]
+
+    def test_init(self):
+        number = NumberWithSeparators("2,000")
+        assert number.number_raw == "2,000"
+        assert isinstance(number.number_raw, str)
+
+    def test_is_valid_ok(self):
+        for num in self.numbers_valid:
+            for sep in self.separators:
+                assert NumberWithSeparators.is_valid(num.replace(",", sep))
+
+    def test_is_valid_wrong(self):
+        for num in self.numbers_wrong:
+            for sep in self.separators:
+                assert not NumberWithSeparators.is_valid(num.replace(",", sep))
+        # Others
+        assert not NumberWithSeparators.is_valid("1.000,213")
+        assert not NumberWithSeparators.is_valid("1.000 213")
+        assert not NumberWithSeparators.is_valid("1")
+        assert not NumberWithSeparators.is_valid("hola")
+
+        with raises(TypeError):
+            assert not NumberWithSeparators.is_valid(1)  # type: ignore
+        with raises(TypeError):
+            assert not NumberWithSeparators.is_valid(1.000)  # type: ignore
+
+    def test_clean_ok(self):
+        for num, num_cor in zip(self.numbers_valid, self.numbers_valid_corrected):
+            number = NumberWithSeparators(num).clean()
+            assert number == num_cor
+
+    def test_clean_wrong(self):
+        for num in self.numbers_wrong:
+            with raises(ValueError):
+                _ = NumberWithSeparators(num).clean()
+
+
+class TestNumberWithWords:
+    def test_init(self):
+        number = NumberWithWords("2 million")
+        assert number.number_raw == "2 million"
+        assert isinstance(number.number_raw, str)
+
+    def test_regex_number_verbose(self):
+        regex = NumberWithWords.regex_number_verbose()
+        assert isinstance(regex, str)
+
+    def test_numeric_words_list(self):
+        words = NumberWithWords.numeric_words_list()
+        assert isinstance(words, set)
+        assert all([isinstance(word, str) for word in words])
+
+    def test_match_numeric_words_ok(self):
+        numbers = {
+            "2 million": {
+                "million": "2",
+                "ten_thousand": 0,
+                "thousand": 0,
+                "hundred": 0,
+                "one": 0,
+            },
+            "2 million 1": {
```

Comment:
Shouldn't it be "2 million and 1" or even better "Two million and one"? I don't know if anyone would write numbers like "2 million 1".

Reply:
I will add a `.replace('and ', '')` so it works when 'and' is used